qiskit/pulse/channels.py

```diff
@@ -35,11 +35,25 @@ class Channel(metaclass=ABCMeta):
     hardware signal channels. The final label -> physical channel mapping is the responsibility
     of the hardware backend. For instance, ``DriveChannel(0)`` holds instructions which the backend
     should map to the signal line driving gate operations on the qubit labeled (indexed) 0.
+
+    To implement a new :class:`Channel` the `prefix` class variable must be set to a string. Which
+    will prefix the channel name when converted to a string. If `prefix` is not set the channel
+    is considered as a base class and may not be instantiated.
```

Comment:
Why does this matter? Start the explanation at a higher level so that even developers who aren't as familiar with the system can follow along.
```suggestion
    Channels are serialized as ``<prefix><index>``. The type of the channel is interpreted from the
    prefix, and the index often maps to the qubit index. All concrete channel classes must have a
    ``prefix`` class attribute (and instances of that class have an index attribute). Base classes have
    ``prefix`` set to ``None``, and are prevented from being instantiated.

    To implement a new :class:`Channel`, provide a unique string identifier for the ``prefix`` class
    attribute.
```

Reply:
Addressed in b3ae47dd3